protocol/nats_jetstream/v3/protocol.go

```diff
@@ -0,0 +1,310 @@
+	}
+
+	if p.url != "" && p.conn != nil {
+		return newValidationError(fieldURL, messageConflictingConnection)
+	}
+
+	consumerConfigOptions := 0
+	if p.consumerConfig != nil {
+		consumerConfigOptions++
+	}
+	if p.orderedConsumerConfig != nil {
+		consumerConfigOptions++
+	}
+
+	if consumerConfigOptions > 1 {
+		return newValidationError(fieldConsumerConfig, messageMoreThanOneConsumerConfig)
+	}
+
+	if len(p.pullConsumeOpts) > 0 && consumerConfigOptions == 0 {
+		return newValidationError(fieldPullConsumerOpts, messageReceiverOptionsWithoutConfig)
+
+	}
+
+	if len(p.publishOpts) > 0 && p.sendSubject == "" {
+		return newValidationError(fieldPublishOptions, messageSenderOptionsWithoutSubject)
+	}
+
+	return nil
+}
+
+// createJetstreamConsumer creates a consumer based on the configured consumer config
+func (p *Protocol) createJetstreamConsumer(ctx context.Context) error {
+	var err error
+	var stream string
+	if stream, err = p.getStreamFromSubjects(ctx); err != nil {
+		return err
+	}
+	var consumerErr error
+	if p.consumerConfig != nil {
+		p.jetstreamConsumer, consumerErr = p.jetStream.CreateOrUpdateConsumer(ctx, stream, *p.consumerConfig)
+	} else if p.orderedConsumerConfig != nil {
+		p.jetstreamConsumer, consumerErr = p.jetStream.OrderedConsumer(ctx, stream, *p.orderedConsumerConfig)
+	} else {
+		return newValidationError(fieldConsumerConfig, messageNoConsumerConfig)
+	}
+	return consumerErr
+}
+
+// getStreamFromSubjects finds the unique stream for the set of filter subjects
+// If more than one stream is found, returns ErrMoreThanOneStream
+func (p *Protocol) getStreamFromSubjects(ctx context.Context) (string, error) {
+	var subjects []string
+	if p.consumerConfig != nil && p.consumerConfig.FilterSubject != "" {
+		subjects = []string{p.consumerConfig.FilterSubject}
+	}
+	if p.consumerConfig != nil && len(p.consumerConfig.FilterSubjects) > 0 {
+		subjects = p.consumerConfig.FilterSubjects
+	}
+	if p.orderedConsumerConfig != nil && len(p.orderedConsumerConfig.FilterSubjects) > 0 {
+		subjects = p.orderedConsumerConfig.FilterSubjects
+	}
+	if len(subjects) == 0 {
+		return "", newValidationError(fieldFilterSubjects, messageNoFilterSubjects)
+	}
+	var finalStream string
+	for i, subject := range subjects {
+		currentStream, err := p.jetStream.StreamNameBySubject(ctx, subject)
+		if err != nil {
+			return "", err
+		}
+		if i == 0 {
+			finalStream = currentStream
+			continue
+		}
+		if finalStream != currentStream {
+			return "", newValidationError(fieldFilterSubjects, messageMoreThanOneStream)
+		}
+	}
+	return finalStream, nil
+}
+
+type msgErr struct {
+	msg binding.Message
+}
+
+// validationError is returned when an invalid option is given
+type validationError struct {
+	field   string
+	message string
+}
+
+// Error returns a message indicating an error condition, with the nil value representing no error.
+func (v validationError) Error() string {
+	return fmt.Sprintf("invalid Spec: %q: %s", v.field, v.message)
```

Comment:
nit: copy paste typo ;)

```suggestion
	return fmt.Sprintf("invalid parameters provided: %q: %s", v.field, v.message)
```